### PR TITLE
feat(annotations): add settings feature

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-1055

Screenshot 2024-04-09 at 12 50 02

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [ ] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Help yourself at 2AM by putting some metrics to help observability

File: libs/shared/ui/src/lib/components/status-chip/status-chip.tsx [hunk 1]
```diff
@@ -50,7 +50,18 @@ export function StatusChip(props: StatusChipProps) {
...
+    .with(
+      'QUEUED',
+      'DELETE_QUEUED',
+      'DEPLOYMENT_QUEUED',
+      'RESTART_QUEUED',
+      'STOP_QUEUED',
+      'WAITING_DELETING',
+      'WAITING_RESTARTING',
+      'WAITING_RUNNING',
+      'WAITING_STOPPING',
```
Comment (RemiBonnet) [inline libs/shared/ui/src/lib/components/status-chip/status-chip.tsx:62]: API doc requirements

File: libs/shared/ui/src/lib/components/navigation-left/navigation-left.tsx [hunk 2]
```diff
@@ -14,7 +15,11 @@ export interface NavigationLeftProps {
 
 export interface NavigationLeftLinkProps {
   title: string
+  /**
+   * @deprecated please use `iconName` instead `icon`
+   */
   icon?: string
+  iconName?: IconName
```
Comment (ctjhoa) [inline libs/shared/ui/src/lib/components/navigation-left/navigation-left.tsx:22]: Add mutual exclusion between `icon` and `iconName` using something like:
```ts
type NavigationLeftLinkProps = {
  // other props
} & (
  | {
      iconName: IconName
    }
  | {
      icon: string
    }
)

```

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/domains/organizations/feature/src/index.ts [hunk 3]
```diff
@@ -8,6 +9,10 @@ export * from './lib/git-token-list/git-token-list'
 export * from './lib/container-registry-create-edit-modal/container-registry-create-edit-modal'
 export * from './lib/hooks/use-add-credit-card/use-add-credit-card'
 export * from './lib/hooks/use-add-credit-code/use-add-credit-code'
+export * from './lib/hooks/use-annotations-groups/use-annotations-groups'
+export * from './lib/hooks/use-create-annotations-group/use-create-annotations-group'
+export * from './lib/hooks/use-edit-annotations-group/use-edit-annotations-group'
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/domains/organizations/feature/src/index.ts]: https://github.com/Qovery/console/pull/1268/commits/58262bf710bf75a4c69be9c17ee33a27353f281e

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**